### PR TITLE
Add net.kuribo64.melonDS

### PR DESCRIPTION
**Pull request by nadiaholmquist**
melonDS is a cross-platform Nintendo DS/DSi emulator, released under the GPLv3.

I'm affiliated with the melonDS project but not the main author, but have got permission to submit and maintain it in Flathub.

### CODE REVIEW

**Comment by nedrichards** [comment]
You don't appear to be actually installing that appdata in the manifest? Ideally it should be in the upstream project and installed as part of the normal course of doing business anyway.

**Comment by nedrichards** [comment]
bot, build net.kuribo64.melonDS

File: net.kuribo64.melonDS.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline net.kuribo64.melonDS.appdata.xml:1]: Should be upstreamed if possible as it's not a flatpak/flathub specific file.

File: net.kuribo64.melonDS.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+<component type="desktop">
+  <id>net.kuribo64.melonDS.desktop</id>
```
Comment (bilelmoussaoui) [inline net.kuribo64.melonDS.appdata.xml:3]:
```suggestion
  <id>net.kuribo64.melonDS</id>
  <launchable type="desktop-id">net.kuribo64.melonDS.desktop</launchable>
```

File: net.kuribo64.melonDS.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	</ul>
+	<p>
+		<strong>Note:</strong> BIOS and firmware files dumped from a real DS or DSi console are required to use melonDS.
+	</p>
+  </description>
+  <url type="homepage">http://melonds.kuribo64.net/</url>
+  <screenshots>
+    <screenshot type="default">https://raw.githubusercontent.com/flathub/net.kuribo64.melonDS/master/screenshots/1.png</screenshot>
+    <screenshot type="default">https://raw.githubusercontent.com/flathub/net.kuribo64.melonDS/master/screenshots/2.png</screenshot>
+    <screenshot type="default">https://raw.githubusercontent.com/flathub/net.kuribo64.melonDS/master/screenshots/3.png</screenshot>
```
Comment (bilelmoussaoui) [inline net.kuribo64.melonDS.appdata.xml:32]: This repo doesn't exists yet and so are the screenshots, the tests will fail for missing/not found screenshots.

File: net.kuribo64.melonDS.appdata.xml (hunk 1 above)
Comment (nadiaholmquist) [reply net.kuribo64.melonDS.appdata.xml:1]: Will look into getting that done in a few days.

File: net.kuribo64.melonDS.appdata.xml (hunk 3 above)
Comment (nadiaholmquist) [reply net.kuribo64.melonDS.appdata.xml:33]: Should I change this for the time being so the tests will pass, or is this fine?

**Comment by nedrichards** [comment]
bot, build net.kuribo64.melonDS

File: net.kuribo64.melonDS.appdata.xml (hunk 3 above)
Comment (barthalion) [reply net.kuribo64.melonDS.appdata.xml:33]: Please change it so we can have the test build pass.

Comment (barthalion) [reply net.kuribo64.melonDS.appdata.xml]: Did that myself, I hope you don't mind.

**Comment by barthalion** [comment]
bot, build net.kuribo64.melonDS

**Comment by barthalion** [comment]
bot, build net.kuribo64.melonDS

**Comment by nadiaholmquist** [comment]
Hmm... I'm not really sure what's up with the aarch64 build failure, if I try building it on my aarch64 system running Arch ARM it works correctly.

**Comment by barthalion** [comment]
No idea. Maybe something in the runtime is miscompiled? @grulja @valentindavid does it ring any bells to you?

Alternatively aarch64 build can be disabled for now: https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64

**Comment by grulja** [comment]
It looks like a build issue in the app itself:
```
src/frontend/qt_sdl/main.h:135:1: error: expected class-name before ‘{’ token
  135 | {
      | ^
In file included from src/frontend/qt_sdl/melonDS_autogen/mocs_compilation.cpp:8:
src/frontend/qt_sdl/melonDS_autogen/EWIEGA46WW/moc_main.cpp: In member function ‘virtual void* ScreenPanelGL::qt_metacast(const char*)’:
src/frontend/qt_sdl/melonDS_autogen/EWIEGA46WW/moc_main.cpp:475:29: error: ‘QOpenGLFunctions_3_2_Core’ does not name a type; did you mean ‘QOpenGLFunctionsPrivate’?

Perhaps the "QOpenGLFunctions_3_2_Core" class is not built for some reason on arm? Missing dependency? 
```

**Comment by barthalion** [comment]
@grulja Seems to be caused by `-opengl es2` at https://invent.kde.org/packaging/flatpak-kde-runtime/-/blob/qt5.15lts/org.kde.Sdk.json.in#L108, at least according to https://gitlab.kitware.com/vtk/vtk/-/issues/17113. Arch ARM does not seem to specify it at all in their PKGBUILD: https://github.com/archlinuxarm/PKGBUILDs/blob/master/extra/qt5-base/PKGBUILD

**Comment by nadiaholmquist** [comment]
Ah well that's unfortunate, currently OpenGL can't be disabled in melonDS (I intend to try fixing that at some point), and it's not able to use GLES.

Sadly the solution might have to be disabling aarch64 for now until this can be fixed upstream.

**Comment by barthalion** [comment]
It's unclear to me whether it's something that should be fixed in KDE SDK's qt build or not. Let's go with disabling aarch64 for now.

**Comment by WaluigiWare64** [comment]
> Ah well that's unfortunate, currently OpenGL can't be disabled in melonDS (I intend to try fixing that at some point), and it's not able to use GLES.

RSDuck added a build option to include the OpenGL renderer (Arisotura/melonDS@697730240394cd9fedcf7924f29ffa625fc784fd).

**Comment by nadiaholmquist** [comment]
Is there a way in the manifest to do platform-specific build options, or should I include a patch for the CMake build script to do it?

**Comment by bilelmoussaoui** [comment]
> Is there a way in the manifest to do platform-specific build options, or should I include a patch for the CMake build script to do it?



> Is there a way in the manifest to do platform-specific build options, or should I include a patch for the CMake build script to do it?

yes, by using build-options
```json
"build-options": {
                "arch": {
                    "aarch64": {
                        "config-opts": [
                            "-DENABLE_OGLRENDERER=OFF"
                        ]
                    }
                }
            },
```

**Comment by nadiaholmquist** [comment]
Well...just realized this doesn't actually matter right now. The stable version (0.9) does not yet have this option.

**Comment by bilelmoussaoui** [comment]
bot, build net.kuribo64.melonDS

**Comment by nadiaholmquist** [comment]
Well that's unfortunate, guess we're gonna have to disable aarch64 then.

**Comment by barthalion** [comment]
bot, build net.kuribo64.melonDS

**Comment by barthalion** [comment]
LGTM. @nadiaholmquist Who besides you should be given write access to Flathub repo?

**Comment by nadiaholmquist** [comment]
Great! I'll probably be the only person maintaining this, but I'd say give @Arisotura and @RSDuck access too.

**Pull request closed without merging**